### ISSUE TITLE
Allow completly blocking remote server

### ISSUE THREAD

**Issue by Findus23** [Chocobozzz/PeerTube#2143]
**What happened?**
In the last days my peertube server seems to getting spam followers.

They all have the same profile picture and are on the same domain (I'll call it `spam.example` as I don't know if it is a legitimate instance).
![grafik](https://user-images.githubusercontent.com/6266037/65391965-cbb11800-dd6f-11e9-98a3-a6eed880003f.png)

`spam.example` seems to be a heavily modified Mastodon instance, but oddly when I search for the users on my mastodon instance, so I can report them, they are not found.
I have now added them to the `Muted servers` list, which seems to have stopped the follower.
But now they still try to follow the mastodon server and show up in the Followers list (with the `Follower handle` as username@spam.example

**To summarize**

- individual users appearing in the notifications are hard to find and thus ban
- individual users can follow the server despite their domains being in the muted servers list

**What do you expect to happen instead?**

Muting a domain should make it impossible for users on this domain to not only follow channels on my peertube server, but also the peertube server itself.

**Comment by pietsch** [comment]
When I deny a follow request, I would like to have an option like "and never allow this domain to ask again".

**Comment by Findus23** [comment]
Thanks @pietsch for reminding me that I seem to have created (and forgotten) this Github issue as this was something that annoyed me again in the last weeks:

I have created a new instance requiring approval for instance followers, but whenever I refuse a follower (as I don't want my content to appear there), they just resend their follow request a day later (maybe even #406 is doing that by default?).

And on another instance where following doesn't require approval, removing a follower is quite pointless as they will just follow again a few days later.

**Comment by pietsch** [comment]
Unfortunately, the problem with re-applying refused instance followers still exists. As this trouble ticket originally described a problem with individual followers, it might be better to create a separate trouble ticket for instance followers.

**Comment by Chocobozzz** [comment]
Hello,

We now remember rejected follow requests so
